python/triton/testing.py

```diff
@@ -320,12 +320,10 @@ def _run(self, bench: Benchmark, save_path: str, show_plots: bool, print_data: b
         y_min = [f'{x}-min' for x in bench.line_names]
         y_max = [f'{x}-max' for x in bench.line_names]
         x_names = list(bench.x_names)
-        df = pd.DataFrame(columns=x_names + y_mean + y_min + y_max)
-        for x in bench.x_vals:
-            # x can be a single value or a sequence of values.
-            if not isinstance(x, (list, tuple)):
-                x = [x for _ in x_names]
-
+        x_vals = [tuple(x for _ in x_names) if not isinstance(x, (list, tuple)) else x for x in bench.x_vals]
+        index = pd.Index(x_vals, name=tuple(x_names))
+        df = pd.DataFrame(index=index, columns=y_mean + y_min + y_max, dtype="float")
```

Comment:
Separating the x and y axis as index and columns respectively seems more correct.